src/storage/scripting.cc

```diff
@@ -1009,12 +1015,30 @@ const char *RedisProtocolToLuaTypeAggregate(lua_State *lua, const char *reply, i
     lua_pushboolean(lua, 0);
     return p;
   }
+  if (atype == '*') {
+    lua_newtable(lua);
+    for (j = 0; j < mbulklen; j++) {
+      lua_pushnumber(lua, j + 1);
+      p = RedisProtocolToLuaType(lua, p);
+      lua_settable(lua, -3);
+    }
+    return p;
+  }
+
+  // atype == '%' or atype == '~'
```

Comment:
I think we need to do some check to make it more stable here, e.g. in the future we have a new type with a new starting character.

Reply:
Yes, that makes a lot of sense. How about doing a check before pushing into lua like:

```
CHECK(atype == '%' || atype == '~')
if (atype == '%' || atype == '~') {
   // do something
}

```